service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -86,6 +87,9 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             caseData.getAllRespondents(), caseDataBefore.getAllRespondents()
         ));
 
+        caseData = getCaseData(caseDetails);
```

Comment:
Don't do this, capture the previous assignment to the map as a variable and then remove from there. i.e.
```java
List<Element<Respondent>> respondents = respondentService.persistRepresentativesRelationship(...);
respondents = respondentService.removeHiddenFields(respondents);
caseDetails.getData().put(RESPONDENTS_KEY, respondents);
```

Reply:
`        List<Element<Respondent>> respondents = respondentService.persistRepresentativesRelationship(
            caseData.getAllRespondents(), caseDataBefore.getAllRespondents());

        caseDetails.getData().put(RESPONDENTS_KEY, respondentService.removeHiddenFields(respondents));`
        did that